File: ios/YamabikoChat/Data/Models/ChatProject.swift
import Foundation
import GRDB

struct ChatProject: Codable, FetchableRecord, MutablePersistableRecord, Identifiable, Equatable {
    static let databaseTableName = "projects"

    var id: Int64?
    var title: String
    var iconName: String
    var colorHex: String
    var instructions: String?
    var createdAtMs: Int64
    var updatedAtMs: Int64

    init(
        id: Int64? = nil,
        title: String,
        iconName: String = "folder.fill",
        colorHex: String = "#3A7AFE",
        instructions: String? = nil,
        createdAtMs: Int64 = Int64(Date().timeIntervalSince1970 * 1000),
        updatedAtMs: Int64 = Int64(Date().timeIntervalSince1970 * 1000)
    ) {
        self.id = id
        self.title = title
        self.iconName = iconName
        self.colorHex = colorHex
        self.instructions = instructions
        self.createdAtMs = createdAtMs
        self.updatedAtMs = updatedAtMs
    }

    mutating func didInsert(_ inserted: InsertionSuccess) {
        id = inserted.rowID
    }
}

File: app/src/main/java/com/porarri/yamabikochat/data/local/ChatProject.kt
package com.porarri.yamabikochat.data.local

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "projects",
    indices = [
        Index(value = ["updatedAtMs"])
    ]
)
data class ChatProject(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val title: String,
    val iconName: String = "folder.fill",
    val colorHex: String = "#3A7AFE",
    val instructions: String? = null,
    val createdAtMs: Long = System.currentTimeMillis(),
    val updatedAtMs: Long = System.currentTimeMillis()
)
